### PR TITLE
[MU4] Add more accidentals

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
see https://musescore.org/en/node/306457 and https://musescore.org/en/node/306456

* add missing Gould arrow down/up accidentals
* add more extended Helmholtz-Ellis accidentals
* add the most important (Spartan) Sagittal accidentals
* add curly braces to accidentals
* (almost) complete MusicXML im- and export

See also #6002 (Add support for triple sharp and triple flat, pending), #5737 (adding Wyschnegradsky accidentals, merged) and #6263 ([MusicXML] Add missing Gould arrow quartertone accidentals, merged)

Counterpart to #6182, which is in 3.x now.

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Boy, what a royal pain to 'cherry-pick' this from 3.x for master :-(

File: libmscore/sym.cpp [hunk 1]
```diff
@@ -2682,10 +2682,12 @@ const std::array<const char*, int(SymId::lastSym) + 1> Sym::symUserNames = { {
     QT_TRANSLATE_NOOP("symUserNames", "No symbol"),
     "4-string tab clef",
     "6-string tab clef",
-    "11 large diesis down, 3\u00b0 down [46 EDO]",
-    "11 large diesis up, (11L), (sharp less 11M), 3\u00b0 up [46 EDO]",
-    "11 medium diesis down, 1\u00b0[17 31] 2\u00b046 down, 1/4-tone down",
-    "11 medium diesis up, (11M), 1\u00b0[17 31] 2\u00b046 up, 1/4-tone up",
+    QT_TRANSLATE_NOOP("symUserNames", "11 large diesis down, 3\u00b0 down [46 EDO]"),
+    QT_TRANSLATE_NOOP("symUserNames", "11 large diesis up, (11L), (sharp less 11M), 3\u00b0 up [46 EDO]"),
+    //QT_TRANSLATE_NOOP("symUserNames", "11 medium diesis down, 1\u00b0[17 31] 2\u00b046 down, 1/4-tone down"),
```
Comment (vpereverzev) [inline libmscore/sym.cpp:2687]: Do we still need this?

Comment (Jojo-Schmitz) [reply libmscore/sym.cpp:2687]: That is the original string as taken from SMuFL, but as such doesn't work with the translations system. So I'd rather keep this (and commented out), as a hint should this string ever change

Comment (Jojo-Schmitz) [reply libmscore/sym.cpp:2687]: Hmm, maybe not...
We'll see once translations got enabled. I do remember there are problems with that `\u` notation and the translations system, if it is not really clear where the number stops and the next 'normal' letter shows

**Review by vpereverzev — APPROVED** [review]

**Pull request merged**